src/main/database.cpp

```diff
@@ -341,7 +342,12 @@ Allocator &Allocator::Get(AttachedDatabase &db) {
 void DatabaseInstance::Configure(DBConfig &new_config, const char *database_path) {
 	config.options = new_config.options;
 
-	if (new_config.options.duckdb_api.empty()) {
+	config.options.unrecognized_options.clear();
```

Comment:
Why are we clearing the map?
Currently we throw if there are unrecognized options

Reply:
That's indeed not necessary - I had a slightly different solution before which used `unrecognized_options` instead of `user_options` but I reworked it as this was cleaner. In any case `unrecognized_options` should always be empty at this point anyway, but I've removed the clear.